src/cephadm/cephadm.py

```diff
@@ -2036,6 +2036,11 @@ def get_fqdn():
     return socket.getfqdn() or socket.gethostname()
 
 
+def get_hostip():
+    # type: () -> str
+    return socket.gethostbyname(socket.gethostname())
```

Comment:
From https://pythontic.com/modules/socket/gethostbyname it seems `gethostbyname` can only return IPv4 addresses. Maybe for IPv6 compatibility, we want to use `getaddrinfo` instead?

Reply:
@liewegas [already embedded this kind of lookup](https://github.com/ceph/ceph/commit/157a7b4183dbd888f106c613a758409d7e07b917) into the generic `mgr.get_mgr_ip()` API call. If the existing call doesn't provide what is needed here, I'd vote for extending it, since this is a pretty common functionality, so IMHO we should go for a common implementation & consistent behaviour across modules.

Reply:
> @liewegas [already embedded this kind of lookup](https://github.com/ceph/ceph/commit/157a7b4183dbd888f106c613a758409d7e07b917) into the generic `mgr.get_mgr_ip()` API call. If the existing call doesn't provide what is needed here, I'd vote for extending it, since this is a pretty common functionality, so IMHO we should go for a common implementation & consistent behaviour across modules.

@epuertat actually this code is executed as part of the `cephadm` binary so there's no access to the mgr code that's why I added this new function. Maybe as part of the refactoring we can put these functions and similar in some utility class.

Reply:
@epuertat I changed the code and changed `gethostname` by `socket.getaddrinfo`  which takes into account ipv6 as well.